### PR TITLE
feat: integrate leverage simulations

### PR DESCRIPTION
**Pull request by toteki**
## Description

closes: #644

----

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added appropriate labels to the PR
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by alexanderbez** [comment]
My hero <3

**Comment by toteki** [comment]
Note: gotta finish this off by adding funds to your random accounts on startup. Currently all simulated operations are working with zero spendable coins, which makes them useless.

Note sure if [this](https://pkg.go.dev/github.com/cosmos/cosmos-sdk/simapp#FundAccount) will do it, but something like that is needed.

File: tests/simulation/state.go [hunk 1]
```diff
@@ -93,6 +114,17 @@ func appStateFn(cdc codec.JSONCodec, simManager *module.SimulationManager) simty
...
+		// add some leverage supported assets to accounts that already have nonzero balances
+		for i, balance := range bankState.Balances {
+			fundedBalance := balance
+			for _, t := range leverageState.Registry {
+				fund := sdk.NewInt64Coin(t.BaseDenom, 100)
+				fundedBalance.Coins = fundedBalance.Coins.Add(fund)
+				bankState.Supply = bankState.Supply.Add(fund)
+			}
+			bankState.Balances[i] = fundedBalance
+		}
```
Comment (toteki) [inline tests/simulation/state.go:126]: this covers the nonzero initial balance issue

**Comment by toteki** [comment]
Note:

Now that nonzero amounts are being used, transactions are failing when asset USD values are needed for tokens that the (simulated) oracle module doesn't have at a given moment.

Need to check oracle sim's initial accept list, then either modify it to contain the three test tokens in leverage simulation or reduce leverage sim to just use `uumee`, than only run transactions that require oracle (borrow, withdraw, liquidate) when the price exists.

File: tests/simulation/state.go [hunk 2]
```diff
@@ -93,6 +114,17 @@ func appStateFn(cdc codec.JSONCodec, simManager *module.SimulationManager) simty
 			panic(err)
 		}
 
+		// add some leverage supported assets to accounts that already have nonzero balances
+		for i, balance := range bankState.Balances {
+			fundedBalance := balance
+			for _, t := range leverageState.Registry {
+				fund := sdk.NewInt64Coin(t.BaseDenom, 100)
```
Comment (toteki) [inline tests/simulation/state.go:121]:
```suggestion
				fund := sdk.NewInt64Coin(t.BaseDenom, 100000000)
```

**Comment by adamewozniak** [comment]
@toteki should we close this for the time being?

**Pull request closed without merging**